pkg/reconciler/ingress/ingress_test.go

```diff
@@ -251,6 +253,36 @@ func TestReconcile(t *testing.T) {
 	}))
 }
 
+func TestKourierIngress(t *testing.T) {
+	createdIng.Status.InitializeConditions()
+	changedService := service(defaultNamespace, testingName)
+	changedService.Spec.ExternalName = "changed"
+	table := TableTest{{
+		Name: "create new ingress with kourier",
+		Key:  "default/testing",
+		Objects: []runtime.Object{
+			ingSometimesAsync,
+		},
+		Ctx:  context.WithValue(context.Background(), "ingressClass", "kourier.ingress.networking.knative.dev"),
+		WantCreates: []runtime.Object{
+			createdIngWithKourier,
+			service(defaultNamespace, testingName),
+		}},
+	}
+
+	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
+			os.Setenv("INGRESS_CLASS_NAME", "kourier.ingress.networking.knative.dev")
```

Comment:
ideally I wouldn't do os.Setenv in the test (inject the value in the context or as a Reconciler field via main.go or the controller instead, which will also decouple the reconciler from config details), but if you do do this here you need to reset it to the old value afterwards or you'll pollute all the other tests if they happen to run after this since the env is global (which is why it's best not to do it 😊).

Reply:
For the moment I have stored and restored the OS variable after the test. 
I will open a new issue to refactor this to include it in context as it seems this may not be using context by default.